docs/DeveloperGuide.md

```diff
@@ -1,34 +1,148 @@
 # Developer Guide
+* [Setting up](#1-setting-up)
+* [Design](#2-design)
+* [Implementation](#3-implementation)
+  * [Undo features](#31-proposed-undo-feature)
+    * [View By Category](#311-view-by-category)
+* [Documentation](#4-documentation)
+* [Testing](#5-testing)
+* [Dev Ops](#6-dev-ops)
+  * [Appendix A: Product Scope](#61-appendix-aproduct-scope)
+  * [Appendix B:Value proposition](#62-appendix-bvalue-proposition)
+  * [Appendix C:User Stories](#63-appendix-cuser-stories)
+  * [Appendix D:Non-Functional Requirements](#64-appendix-dnon-functional-requirements)
+  * [Appendix E:Glossary](#65-appendix-eglossary)
+  * [Appendix F:Instructions for Manual Testing](#66-appendix-finstructions-for-manual-testing)
+## 1. Setting up
+## 2. Design
 
-## Design & Implementation
+{Describe the design of the product. Use **Architecture Diagram** which has not been covered yet.}
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 3. Implementation
+### 3.1 [Proposed] Features
+#### 3.1.1 List By Category
+It extends the current view command.
+The user can use `list` to view all tasks.
+The user can also use `list CATEGORY` to view specific classes whose category is sth. 
+This is the UML design for view by category.
+![UML for View](images/viewCategory.png)
```

Comment:
![command abstract](https://user-images.githubusercontent.com/42445973/77891822-250c7d80-72a4-11ea-85f3-348201a5a742.PNG)
Should the `{abstract}` label be placed on top of the `Command` class name, instead of being placed on its right?